File: Backend/login-microsoft365/src/main/java/com/miapp/model/Editorial.java
package com.miapp.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "EDITORIAL")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Editorial {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String autorPersonal;
    private String autorSecundario;
    private String autorInstitucional;
    private String editorial;
    private String coordinador;
    private String director;
    private String compilador;
    private String serie;
    private String descripcionFisica;
    private Integer cantidad;
    private Integer anioPublicacion;
    private String edicion;
    private String reimpresion;
    private String isbn;

    // Relaciones con Pais, Ciudad, Idioma (asumiendo que existen)
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CODIGOPAIS", nullable = false)
    private Pais pais;

    @ManyToOne
    @JoinColumn(name = "CODIGOCIUDAD", referencedColumnName = "CODIGOCIUDAD")
    private Ciudad ciudad;

    @ManyToOne
    @JoinColumn(name = "idioma_id")
    private Idioma idioma;

    @OneToOne
    @JoinColumn(name = "material_bibliografico_id")
    @JsonBackReference  // No serializa la parte del material para evitar recursión
    private MaterialBibliografico materialBibliografico;
}

File: Frontend/sakai-ng-master/src/app/biblioteca/interfaces/material-bibliografico/editorial.ts
import { EscuelaProfesional } from "../escuela-profesional";


export class Editorial {
    id: number;
    autorPrincipal:string;
    autorSecundario:string;
    autorInstitucional:string;
    editorial:string;
    coordinador:string;
    director:string;
    compilador:string;
    pais:any;
    ciudad:any;
    idioma:any;
    serie:string;
    descripcionFisica:any;
    cantidad:number;
    anioPublicacion:any;
    anio:string;
    edicion:any;
    reimpresion:any;
    isbn:any;
    issn:any;


    constructor(init?: Partial<Editorial>) {
        this.id = 0;
        this.autorPrincipal='';
        this.autorSecundario='';
        this.autorInstitucional='';
        this.editorial='';
        this.coordinador='';
        this.director='';
        this.compilador='';
        this.pais=null;
        this.ciudad=null;
        this.idioma=null;
        this.serie='';
        this.descripcionFisica=null;
        this.cantidad=0;
        this.anioPublicacion=null;
        this.anio='';
        this.edicion=null;
        this.reimpresion=null;
        this.isbn=null;
        this.issn=null;
        // Inicialización opcional si se pasa un objeto
        Object.assign(this, init);
    }
  }
